Sources/SpeziLLM/LLMRunner.swift

```diff
@@ -8,192 +8,187 @@
-    ///   - dependencies: A result builder that aggregates all stated ``LLMRunnerSetupTask``'s as dependencies.
+    ///   - dependencies: A result builder that aggregates all stated ``LLMPlatform``s.
     public init(
-        runnerConfig: LLMRunnerConfiguration = .init(),
-        lazyRunnerSetup: Bool = true,
-        @LLMRunnerSetupTaskBuilder _ dependencies: @Sendable () -> DependencyCollection
+        @LLMPlatformBuilder _ dependencies: @Sendable () -> DependencyCollection
     ) {
-        self.runnerConfiguration = runnerConfig
-        self.lazyRunnerSetup = lazyRunnerSetup
-        self._runnerSetupTaskModules = Dependency(using: dependencies())
-        
-        for modelType in LLMHostingType.allCases {
-            self.runnerBackendInitialized[modelType] = false
-        }
+        self._llmPlatformModules = Dependency(using: dependencies())
     }
     
-    /// Convenience initializer for the creation of a ``LLMRunner``.
+    /// Convenience initializer for the creation of an ``LLMRunner`` that doesn't support any ``LLMPlatform``s
+    /// Helpful for stating a Spezi `Dependency` to the ``LLMRunner``.
     public init() {
-        self.init(runnerConfig: .init()) {}
+        self.init {}
     }
     
+    
     public nonisolated func configure() {
         Task {
-            await mapRunnerSetupTasks()
+            await mapLLMPlatformModules()
         }
     }
     
-    private func mapRunnerSetupTasks() async {
-        for module in runnerSetupTaskModules {
-            guard let task = module as? any LLMRunnerSetupTask else {
-                preconditionFailure("SpeziLLM: Reached inconsistent state. \(type(of: module)) is not a \((any LLMRunnerSetupTask).self)")
-            }
-            
-            runnerSetupTasks[task.type] = task
-            
-            if !lazyRunnerSetup {
-                try? await task.setupRunner(runnerConfig: self.runnerConfiguration)
-                runnerBackendInitialized[task.type] = true
-            }
+    /// Turns the received ``LLMSchema`` to an executable ``LLMSession``.
+    ///
+    /// The ``LLMRunner`` uses the configured ``LLMPlatform``s to create an executable ``LLMSession`` from the passed ``LLMSchema``
+    ///
+    /// - Parameters:
+    ///   - with: The ``LLMSchema`` that should be turned into an ``LLMSession``.
+    ///
+    /// - Returns: The ready to use ``LLMSession``.
+    public func callAsFunction<L: LLMSchema>(with llmSchema: L) async -> L.Platform.Session {
+        // Searches for the respective `LLMPlatform` associated with the `LLMSchema`.
+        guard let platform = llmPlatforms[ObjectIdentifier(L.self)] else {
+            preconditionFailure("""
+            The designated `LLMPlatform` to run the `LLMSchema` was not configured within the Spezi `Configuration`.
+            Ensure that the `LLMRunner` is set up with all required `LLMPlatform`s
```

Comment:
You could even inline the name of the Scheme using `String(describing: ...)` to make the error messages even more descriptive as you do below.